nilearn/regions/signal_extraction.py

```diff
@@ -15,11 +15,166 @@
 from .. import masking
 from ..image import new_img_like
 
+INF = 1000 * np.finfo(np.float32).eps
+
+
+def _check_shape_compatibility(img1, img2, dim=3):
+    """Check that shapes match for dimensions going from 0 to dim-1."""
+    if img1.shape[:dim] != img2.shape[:dim]:
+        raise ValueError("Images have incompatible shapes.")
+
+
+def _check_affine_equality(img1, img2):
+    if (
+        img1.affine.shape != img2.affine.shape
+        or abs(img1.affine - img2.affine).max() > INF
+    ):
+        raise ValueError("Images have different affine matrices.")
+
+
+def _check_shape_affine_label_img(target_img, labels_img):
+    """Validate shapes and affines of labels.
+
+    Check that the provided target and labels images:
+        - have the same shape
+        - have the same affine matrix.
+
+    Parameters
+    ----------
+    target_img : Niimg-like object
+        See :ref:`extracting_data`.
+        Image to extract the data from.
+
+    labels_img : Niimg-like object
+        See :ref:`extracting_data`.
+        Regions definition as labels.
+        Encodes the region labels of the signals.
+
+    """
+    _check_shape_compatibility(target_img, labels_img, dim=3)
+    _check_affine_equality(target_img, labels_img)
+
+
+def _check_shape_affine_maps_masks(target_img,
+                                   mask_img=None,
```

Comment:
We don't care whether the second image is a mask image or anything else. The function could be simply '_check_shape_affine_maps'

Reply:
Does it need to be different from _check_shape_affine_label_img ?

Reply:
I agree, in my initial comment functions were generic. 
I'm probably missing something but I'm not sure to understand why you need to differentiate the behavior of shape and affine checking for masks and label images.

Reply:
tried to use only a single function for both cases